node/modules/actorevent.go

```diff
@@ -91,61 +124,57 @@ func EthEventAPI(cfg config.FevmConfig) func(helpers.MetricsCtx, repo.LockedRepo
 			})
 		}
 
-		ee.EventFilterManager = &filter.EventFilterManager{
+		fm := &filter.EventFilterManager{
 			ChainStore: cs,
 			EventIndex: eventIndex, // will be nil unless EnableHistoricFilterAPI is true
+			// TODO:
+			// We don't need this address resolution anymore once https://github.com/filecoin-project/lotus/issues/11594 lands
 			AddressResolver: func(ctx context.Context, emitter abi.ActorID, ts *types.TipSet) (address.Address, bool) {
-				// we only want to match using f4 addresses
 				idAddr, err := address.NewIDAddress(uint64(emitter))
 				if err != nil {
 					return address.Undef, false
 				}
 
 				actor, err := sm.LoadActor(ctx, idAddr, ts)
 				if err != nil || actor.Address == nil {
-					return address.Undef, false
+					return idAddr, false
 				}
 
-				// if robust address is not f4 then we won't match against it so bail early
-				if actor.Address.Protocol() != address.Delegated {
-					return address.Undef, false
-				}
-				// we have an f4 address, make sure it's assigned by the EAM
-				if namespace, _, err := varint.FromUvarint(actor.Address.Payload()); err != nil || namespace != builtintypes.EthereumAddressManagerActorID {
-					return address.Undef, false
-				}
 				return *actor.Address, true
 			},
 
 			MaxFilterResults: cfg.Events.MaxFilterResults,
 		}
-		ee.TipSetFilterManager = &filter.TipSetFilterManager{
-			MaxFilterResults: cfg.Events.MaxFilterResults,
-		}
-		ee.MemPoolFilterManager = &filter.MemPoolFilterManager{
-			MaxFilterResults: cfg.Events.MaxFilterResults,
-		}
 
 		lc.Append(fx.Hook{
 			OnStart: func(context.Context) error {
 				ev, err := events.NewEvents(ctx, &evapi)
 				if err != nil {
 					return err
 				}
-				// ignore returned tipsets
-				_ = ev.Observe(ee.EventFilterManager)
-				_ = ev.Observe(ee.TipSetFilterManager)
-
-				ch, err := mp.Updates(ctx)
-				if err != nil {
-					return err
-				}
-				go ee.MemPoolFilterManager.WaitForMpoolUpdates(ctx, ch)
-
+				_ = ev.Observe(fm)
 				return nil
 			},
 		})
 
+		return fm, nil
+	}
+}
+
+func ActorEventAPI(cfg config.FevmConfig) func(helpers.MetricsCtx, repo.LockedRepo, fx.Lifecycle, *filter.EventFilterManager, *store.ChainStore, *stmgr.StateManager, EventAPI, *messagepool.MessagePool, full.StateAPI, full.ChainAPI) (*full.ActorEvent, error) {
+	return func(mctx helpers.MetricsCtx, r repo.LockedRepo, lc fx.Lifecycle, fm *filter.EventFilterManager, cs *store.ChainStore, sm *stmgr.StateManager, evapi EventAPI, mp *messagepool.MessagePool, stateapi full.StateAPI, chainapi full.ChainAPI) (*full.ActorEvent, error) {
+		ee := &full.ActorEvent{
```

Comment:
Out of curiosity, what happens if we just return `nil, nil` (or even `nil, error`) if this method is called with `EnableActorEventsAPI = false`? 

That _seems_ more correct to me, since there's just no functionality here if `EnableActorEventsAPI` is false, and so this should never be called in the first place that way. But I'm not sure whether that breaks something?

Regardless, this seems correct as written.

Reply:
@arajasek Actually, this is never called if `EnableActorEventsAPI` if false

See

```
ApplyIf(isFullNode,
			If(cfg.Fevm.EnableActorEventsAPI,
				Override(new(full.ActorEventAPI), modules.ActorEventAPI(cfg.Fevm)),
			),
			If(!cfg.Fevm.EnableActorEventsAPI,
				Override(new(full.ActorEventAPI), &full.ActorEventDummy{}),
			),
		),
```

I think this is just a nice to have sanity check.

Reply:
I agree, but in that case I think this method should just fail if called this way. But it doesn't matter too much.

Reply:
Punting this for now in the first draft.